### PR TITLE
All kubernetes deployments now network host

### PR DESCRIPTION
**Pull request by mhl787156**
This PR moves the simulation of Gazebo and PX4 SITL to run on host network (192.168.xxx.xxx) instead of on kube network (10.24.xxx.xxx). This ensures transparency between all ros nodes (solving Issue #49 ) on all nodes without requiring DDS discovery servers. The changes include the following:

1. Gazebo and px4 sitl deployments now have `networkhost:true` Gazebo now no longer requires the services and displays its own used ports. Px4 deployment px4_sim_host is now localhost and is successful at connecting. 
2. By default if ROS DDS detects two nodes are on the same network, they will use Shared Memory Transport to exchange messages. As the two nodes are not in the same namespace (different pods) this causes communication to fail. Therefore a custom FastRTPS profile is added to all base docker images which deal with ROS: base controller, base gazebo sim, mavros, UI.

### CODE REVIEW

**Comment by rob-clarke** [comment]
It's probably worth keeping an eye on https://github.com/eProsima/Fast-DDS/pull/1801 which, as I understand it, adds some additional ways for FastDDS to distinguish hosts.

This should also allow multiple containers to work without using the shared process namespace thanks to the random number added to the process ID. Adding the process namespace might have a similar effect?

File: simulator/vehicles/iris/xacro_launch.sh [hunk 1]
```diff
@@ -17,7 +17,8 @@ echo "Specific SDF saved to /tmp/${PX4_SIM_MODEL}_${PX4_SYSID}.sdf"
 # Drones will start in a spiral starting from (PX4_SIM_INIT_LOC_X, PX4_SIM_INIT_LOC_Y)
 re_number='^[0-9]+$'
 ORDINAL="${HOSTNAME##*-}"
-if [[ $ORDINAL =~ $re_number && ! ${PX4_SIM_FORCE_USE_SET_POSITION} ]] ; then
+if [[ $ORDINAL =~ $re_number ]] && ! ${PX4_SIM_FORCE_USE_SET_POSITION} ; then
+    echo "Startling location set by spiral"
```
Comment (rob-clarke) [inline simulator/vehicles/iris/xacro_launch.sh:21]: Typo
```suggestion
    echo "Starting location set by spiral"
```

File: simulator/base/core/Dockerfile [hunk 2]
```diff
@@ -23,7 +23,9 @@ RUN . /etc/os-release \
     && echo "deb http://packages.osrfoundation.org/gazebo/$ID-stable `lsb_release -sc` main" > /etc/apt/sources.list.d/gazebo-latest.list
 
 # install gazebo packages
-RUN apt-get update && apt-get install -y --no-install-recommends \
+RUN apt-get update && apt-get install -y \
+    libgazebo11=11.3.0-1* \
+    gazebo11-common=11.3.0-1*\
     gazebo11=11.3.0-1* \
```
Comment (rob-clarke) [inline simulator/base/core/Dockerfile:29]: The better option here might be to remove the explicit version dependency. This does make the build a little less repeatable but I think it fixes this issue. Seems something odd happens with OSRF's package releases

File: deployment/k8.gazebo-iris.amd64.yaml [hunk 3]
```diff
@@ -108,4 +34,32 @@ spec:
...
+        - containerPort: 7419
+          protocol: UDP
+        - containerPort: 7422
+          protocol: UDP
+        - containerPort: 7423
+          protocol: UDP
+        - containerPort: 10318
+          protocol: UDP
+        - containerPort: 10317
+          protocol: UDP
```
[the remark this answers is not in the document]
Comment (mhl787156) [reply deployment/k8.gazebo-iris.amd64.yaml:64]: Hmm interesting I was looking at the kube API [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#podspec-v1-core) where networkHost says "Host networking requested for this pod. Use the host's network namespace. If this option is set, the ports that will be used must be specified. Default to false." So I specified them. Not sure how they are used practically.

File: simulator/base/core/Dockerfile (hunk 2 above)
Comment (mhl787156) [reply simulator/base/core/Dockerfile:29]: Okay I'll give that a try

**Review by rob-clarke — APPROVED** [review]
We may have to keep an eye on the versioning if it breaks the link with `gzweb`, though that had to be modified anyway.

Also, this still results in a massive image for Gazebo. I did start to look at compiling from source to try to exclude the client as we're using `gzweb` instead but I didn't get very far. The build process didn't seem very modular.

Both issues for the future though...

**Pull request merged**